[Bug]: flickering issue resurfaced

### Problem encountered

ST failed because `average entry price` is of a different precision than what is expected.

```
>           assert len(event["updates"]["positions"][0]["averageEntryPrice"]) == expected_average_entry_price_length
E           AssertionError: assert 1 == 10
E            +  where 1 = len('0')
```

### Observed behaviour

The test was put in place to catch issues that cause price flickering as per https://github.com/vegaprotocol/vega/issues/10036


### Expected behaviour

average entry price should be in market price

### Steps to reproduce

```bash
1. run the test `test_average_entry_price_stream`
```


### Software version

dev branch

### Failing test

test_average_entry_price_stream

### Jenkins run

https://jenkins.vega.rocks/job/common/job/system-tests-nightly-perpetuals/32/testReport/junit/tests.API/grpc_streams_test/Call_tests___full_a_d___test_average_entry_price_stream/

### Configuration used

_No response_

### Relevant log output

_No response_

with some additional logging it looks like the the api is not filtering on party 

The party and market are provided in the test

https://jenkins.vega.rocks/job/common/job/system-tests-wrapper/59018/testReport/junit/tests.API/grpc_streams_test/test_average_entry_price_stream/

`AssertionError: expected avp for party f14a7787f79ab4303e2f257f67434b7c8065f0d49f094d8d30a36e63cfe7d866 on market f88036973fe892bf2ee14291c09ffcf4212c88ffcdeecd3b3083cdc14b5bc951 to be in market decimals {'updates': {'positions': [{'marketId': 'f88036973fe892bf2ee14291c09ffcf4212c88ffcdeecd3b3083cdc14b5bc951', 'partyId': '1f1cce03c203c60402ca46b443e84d9c53d602fd46b3df2a27e57b14be74a96b', 'realisedPnl': '0', 'unrealisedPnl': '0', 'averageEntryPrice': '0', 'updatedAt': '1716206874474083000', 'lossSocialisationAmount': '0'}, {'marketId': 'f88036973fe892bf2ee14291c09ffcf4212c88ffcdeecd3b3083cdc14b5bc951', 'partyId': 'f14a7787f79ab4303e2f257f67434b7c8065f0d49f094d8d30a36e63cfe7d866', 'openVolume': '10000', 'realisedPnl': '0', 'unrealisedPnl': '0', 'averageEntryPrice': '9660000000', 'updatedAt': '1716206874474083000', 'lossSocialisationAmount': '0'}, {'marketId': 'f88036973fe892bf2ee14291c09ffcf4212c88ffcdeecd3b3083cdc14b5bc951', 'partyId': '69516e26b1d8bb7151159e0cf37fc857ec7e69831ec6f1ba1a06027431acaf96', 'openVolume': '-10000', 'realisedPnl': '0', 'unrealisedPnl': '0', 'averageEntryPrice': '9660000000', 'updatedAt': '1716206874474083000', 'lossSocialisationAmount': '0'}]}}`

--- a/datanode/api/trading_data_v2.go
+++ b/datanode/api/trading_data_v2.go
@@ -1604,7 +1604,7 @@ func (t *TradingDataServiceV2) ObservePositions(req *v2.ObservePositionsRequest,
 	}
 
 	// add the party to the derived parties
-	if len(derivedParties) > 0 {
+	if req.PartyId != nil && len(*req.PartyId) > 0 {
 		derivedParties = append(derivedParties, *req.PartyId)
 	}
 

